### PR TITLE
ref: Use different error message for empty strings in schema processing

### LINKED ISSUE #2137: "expected a non-empty value" is a super ambiguous error message for empty strings
I recommend we rephrase this to something like "expected a non-empty string". This is what it looks like in the interface:

Screenshot 2023-05-19 at 15 12 10

Took me some JSON digging to understand what Sentry wants me to do.

https://github.com/getsentry/relay/blob/6f2946c479a44e96722c95ccf47b5985ce4e5dd5/relay-general/src/types/meta.rs#L314

### PR DESCRIPTION
**Pull request by lforst**
Fixes #2137

### CODE REVIEW

File: relay-general/src/store/schema.rs [hunk 1]
```diff
@@ -139,9 +155,35 @@ mod tests {
...
+                bar: Annotated::from_error(Error::expected("a non-empty string"), None),
+                bar2: Annotated::new(String::default())
+            })
+        );
+    }
+
     #[test]
     fn test_nonempty_string() {
-        assert_nonempty_base::<String>();
+        assert_nonempty_string::<String>();
```
Comment (jjbayer) [inline relay-general/src/store/schema.rs]: Can we make `assert_nonempty_base` expect a `&str` with the expected error message instead of duplicating the function?

**Review by jjbayer — APPROVED** [review]
See comment about test, other than that, LGTM!

File: relay-general/src/store/schema.rs (hunk 1 above)
Comment (lforst) [reply relay-general/src/store/schema.rs]: Great feedback. Thanks! 29a689ef7cdfa2f3c3d05aff44626cb77a7067b6
